### PR TITLE
Add open interest to market data

### PR DESCRIPTION
**Pull request by campbellssource**
Should open interest be added here? does it make sense as a core API?

### CODE REVIEW

File: specs/0021-market-data-spec.md [hunk 1]
```diff
@@ -10,7 +10,8 @@ Specification PR: https://gitlab.com/vega-protocol/product/merge_requests/22
 - [ ] The mid price is empty / nothing if there is either no buy order or no sell orders.
 - [ ] The mid price is the arithmetic average of the best bid price and best offer price.
 - [ ] The mark price calculation uses the methodology specified in the market framework.
-
+- [ ] The Open interest returns the sum for all open positions where positions size is greater than 0.
+- [ ] The Open interest returns 0 if there are no positions on the market
```
Comment (tamlyn10) [inline specs/0021-market-data-spec.md:15]: I'd make this slightly more explicit by saying 

The Open interest returns **the sum of the size** of all open positions where positions size is greater than 0.

File: specs/0021-market-data-spec.md [hunk 2]
```diff
@@ -31,6 +32,7 @@ All these values can be empty/nothing if there is insufficient relevant data.
   - **Best offer volume:** the aggregated volume being offered at the _best offer price_.
   - **Mid price:** the arithmetic average of the _best bid price_ and _best offer price_.
   - **Mark price:** the current mark price as calculated by the selected mark price methodology.
+  - **Open interest:** the sum of all positions above 0.
```
Comment (tamlyn10) [inline specs/0021-market-data-spec.md]: as per above comment

**Review by tamlyn10 — CHANGES_REQUESTED** [review]
Suggested making definition of open interest more explicit

**Review by tamlyn10 — APPROVED** [review]
Needs a member of core team to approve.

**Comment by tamlyn10** [comment]
@jeremyletang are  you able to approve this?

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**